src/pages/ProfilesExplorerView/components/SceneExploreServiceFlameGraph/components/SceneFunctionDetailsPanel/components/GitHubContextProvider/GitHubContextProvider.tsx

```diff
@@ -1,24 +1,38 @@
 import { displayError } from '@shared/domain/displayStatus';
-import React, { useCallback, useState } from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import { DataSourceProxyClientBuilder } from 'src/pages/ProfilesExplorerView/infrastructure/series/http/DataSourceProxyClientBuilder';
 
+import { VcsClient } from '../../infrastructure/VcsClient';
 import { generateNonce } from './domain/generateNonce';
 import { githubLogin } from './domain/githubLogin';
 import { useGithubSessionCookie } from './domain/useGithubSessionCookie';
 import { usePollGitHubPopup } from './domain/usePollGitHubPopup';
 import { DEFAULT_GITHUB_CONTEXT, GitHubContext } from './GitHubContext';
+import { PrivateVcsClient } from './infrastructure/PrivateVcsClient';
 
 type GitHubContextProviderProps = {
+  dataSourceUid: string;
   children: React.ReactNode;
 };
 
 export const nonce = generateNonce();
 
-export function GitHubContextProvider({ children }: GitHubContextProviderProps) {
+export function GitHubContextProvider({ dataSourceUid, children }: GitHubContextProviderProps) {
+  const vcsClient = DataSourceProxyClientBuilder.build(dataSourceUid, VcsClient);
+  const privateVcsClient = DataSourceProxyClientBuilder.build(dataSourceUid, PrivateVcsClient);
+
   const [isLoginInProgress, setIsLoginInProgress] = useState(DEFAULT_GITHUB_CONTEXT.isLoginInProgress);
   const [sessionCookie, setSessionCookie] = useGithubSessionCookie();
   const [externalWindow, setExternalWindow] = useState<Window | null>();
 
-  usePollGitHubPopup({ externalWindow, setExternalWindow, setSessionCookie, nonce });
+  // hack to prevent failures impossible to fix for the user (unless they know they have to delete the cookie)
+  // when logged in and changing data source
+  // TODO: provide a better way
+  useEffect(() => {
+    setSessionCookie('');
+  }, [dataSourceUid]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
This requires users to click "Connect to ..." every time the reload the page, even if they're logged in, right?

Certainly I've observed that if I reload the page, I have to click the button again. Not a blocker to this PR, but definitely mildly annoy.

<img width="840" alt="Screenshot 2024-08-28 at 4 23 41 PM" src="https://github.com/user-attachments/assets/c94c000a-4c21-4c6c-a716-e4e3929e7243">